### PR TITLE
initialize env with abspath

### PR DESCRIPTION
**Pull request by dlage**
When running scripts, the path used is relative to the web2py root instead of the full path. This causes differences in migrations when using InDBMigrator - this causes pydal to try to repeat migrations already done.

### CODE REVIEW

**Comment by mdipierro** [comment]
Can you explain the use case better? It seems to me it is the opposite. If the project is moved and you use the absolute path, it will trigger unwanted migrations.

**Comment by dlage** [comment]
The issue is that a different path is used when running scripts in shell or when a web request is received. When using file storage it's really the same but when we use InDBMigrator each environment would follow a different pattern creating migration issues.

I agree that it would be better to use relative paths but since the (I believe mostly used way to migrate) is using the web-based option (since the --force_migrate option hasn't even been released) I thought this would cause less issues with current deploys (which is what worries me the most).

File: gluon/shell.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ def run(
     errmsg = 'invalid application name: %s' % appname
     if not a:
         die(errmsg, error_preamble=False)
-    adir = os.path.join('applications', a)
+    adir = os.path.absdir(os.path.join('applications', a))
```
Comment (mdipierro) [inline gluon/shell.py:237]: OK. but should be os.path.abspath(....)

**Comment by mdipierro** [comment]
Sorry for the very late reply.

**Comment by mdipierro** [comment]
Sorry this took so long.

**Comment by leonelcamara** [comment]
@mdipierro you forgot your own fix :) this should be abspath

**Comment by mdipierro** [comment]
LOL

**Pull request merged**